src/lksearch/catalog.py

```diff
@@ -0,0 +1,399 @@
+            .set_index("ID")
+        )
+
+    result = result[catalog_name]
+
+    # Rename the columns so that the output is uniform
+    result.rename_columns(
+        catalog_meta["rename_in"],
+        catalog_meta["rename_out"],
+    )
+
+    if catalog_meta["prefix"] is not None:
+        prefix = catalog_meta["prefix"]
+        result["ID"] = [f"{prefix} {id}" for id in result["ID"]]
+
+    # Based on the input coordinates pick the object with the mininmum separation as the reference star.
+    c1 = SkyCoord(result["RAJ2000"], result["DEJ2000"], unit="deg")
+
+    sep = coord.separation(c1)
+
+    # Find the object with the minimum separation - this is our target
+    ref_index = np.argmin(sep)
+
+    # apply_propermotion
+    result = _apply_propermotion(result, equinox=catalog_meta["equinox"], epoch=epoch)
+
+    # Now we want to repete but using the values corrected for proper motion
+    # First get the correct values for target
+    coord_pm_correct = SkyCoord(
+        result["RA"][ref_index], result["DEC"][ref_index], unit="deg"
+    )
+
+    c1_pm_correct = SkyCoord(result["RA"], result["DEC"], unit="deg")
+
+    # Then calculate the separation based on pm corrected values
+    sep_pm_correct = coord_pm_correct.separation(c1_pm_correct)
+
+    # Provide the separation in the output table
+    result["Separation"] = sep_pm_correct
+
+    # Calculate the relative flux
+    result["Relative_Flux"] = 10 ** (
+        (
+            [value for key, value in result.items() if "_Mag" in key][0]
+            - [value for key, value in result.items() if "_Mag" in key][0][ref_index]
+        )
+        / -2.5
+    )
+
+    # Now sort the table based on separation
+    result.sort(["Separation"])
+    return result.to_pandas().set_index("ID")
+
+
+def query_KIC(
+    coord: SkyCoord,
+    epoch: Time,
+    radius: Union[float, u.Quantity] = u.Quantity(100, "arcsecond"),
+    magnitude_limit: float = 18.0,
+):
+    """Query the Kepler Input Catalog for a single source location, obtain nearby sources
+
+    Parameters:
+    -----------
+    coord : astropy.coordinates.SkyCoord
+        Coordinates around which to do a radius query
+    epoch: astropy.time.Time
+        The time of observation in JD.
+    catalog: str
+        The catalog to query, either 'kepler', 'k2', or 'tess', 'gaia'
+    radius : float or astropy quantity
+        Radius in arcseconds to query
+    magnitude_limit : float
+        A value to limit the results in based on the Tmag/Kepler mag/K2 mag or Gaia G mag. Default, 18.
+    Returns:
+    -------
+    result: pd.DataFrame
+        A pandas dataframe of the sources within radius query, corrected for proper motion
+    """
+    if radius.unit == u.pixel:
+        radius = (radius * (4 * u.arcsecond / u.pixel)).to(u.arcsecond)
+    return query_catalog(
+        coord=coord,
+        epoch=epoch,
+        catalog="kic",
+        radius=radius,
+        magnitude_limit=magnitude_limit,
+    )
+
+
+def query_TIC(
```

Comment:
Do we want to have query functions for each catalog, or as a parameter in the main call, like query_catalog(catalog='TIC')